core/domain/user_id_migration.py

```diff
@@ -0,0 +1,515 @@
+        """Return a list of datastore class references to map over."""
+        return [collection_models.CollectionRightsSnapshotContentModel,
+                exp_models.ExplorationRightsSnapshotContentModel,
+                question_models.QuestionRightsSnapshotContentModel,
+                skill_models.SkillRightsSnapshotContentModel,
+                topic_models.TopicRightsSnapshotContentModel]
+
+    @staticmethod
+    def map(rights_snapshot_model):
+        """Implements the map function for this job."""
+        class_name = rights_snapshot_model.__class__.__name__
+        try:
+            if isinstance(
+                    rights_snapshot_model,
+                    collection_models.CollectionRightsSnapshotContentModel):
+                SnapshotsUserIdMigrationJob._migrate_collection(
+                    rights_snapshot_model)
+            elif isinstance(
+                    rights_snapshot_model,
+                    exp_models.ExplorationRightsSnapshotContentModel):
+                SnapshotsUserIdMigrationJob._migrate_exploration(
+                    rights_snapshot_model)
+            elif isinstance(
+                    rights_snapshot_model,
+                    question_models.QuestionRightsSnapshotContentModel):
+                SnapshotsUserIdMigrationJob._migrate_question(
+                    rights_snapshot_model)
+            elif isinstance(
+                    rights_snapshot_model,
+                    skill_models.SkillRightsSnapshotContentModel):
+                SnapshotsUserIdMigrationJob._migrate_skill(
+                    rights_snapshot_model)
+            elif isinstance(
+                    rights_snapshot_model,
+                    topic_models.TopicRightsSnapshotContentModel):
+                SnapshotsUserIdMigrationJob._migrate_topic(
+                    rights_snapshot_model)
+        except MissingUserException as e:
+            yield ('FAILURE - %s' % class_name, e)
+        else:
+            yield ('SUCCESS - %s' % class_name, rights_snapshot_model.id)
+
+    @staticmethod
+    def reduce(key, ids):
+        """Implements the reduce function for this job."""
+        if key.startswith('SUCCESS'):
+            yield (key, len(ids))
+        else:
+            yield (key, ids)
+
+
+class GaeIdNotInModelsVerificationJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for going through all the UserSettingsModels and checking
+    that the gae_id is not mentioned in any of the fields that should contain
+    user_id, this job also checks that all the new user IDs are 32 lowercase
+    chars long strings.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Return a list of datastore class references to map over."""
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(user_model):
+        """Implements the map function for this job."""
+        gae_id = user_model.gae_id
+        if (len(user_model.id) != 32 or
```

Comment:
Move 32 into a named constant and use this named constant in the relevant places.

Reply:
Done, the constant already existed forgot to use it here.